dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataSourceService.java

```diff
@@ -499,7 +499,13 @@ public String buildParameter(DbType type, String host,
 
         String address = buildAddress(type, host, port, connectType);
         Map<String, Object> parameterMap = new LinkedHashMap<String, Object>(6);
-        String jdbcUrl = address + "/" + database;
+        String jdbcUrl;
+        if (DbType.SQLSERVER == type) {
+            jdbcUrl = address + ";databaseName=" + database;
+        } else {
+            jdbcUrl = address + "/" + database;
+        }
+
         if (Constants.ORACLE.equals(type.name())) {
```

Comment:
Do you mind removing the following constants? I think using '==' is better, after all, the 'type' variable is Enum

    public static final String MYSQL = "MYSQL";
    public static final String POSTGRESQL = "POSTGRESQL";
    public static final String HIVE = "HIVE";
    public static final String SPARK = "SPARK";
    public static final String CLICKHOUSE = "CLICKHOUSE";
    public static final String ORACLE = "ORACLE";
    public static final String SQLSERVER = "SQLSERVER";
    public static final String DB2 = "DB2";
    public static final String PRESTO = "PRESTO";

Reply:
agree with @gabrywu idea . but this modify maybe involved not only DbType . suggest again submit a new PR